spec/core/matchers/matchersUtilSpec.js

```diff
@@ -613,6 +639,26 @@ describe("matchersUtil", function() {
       testFunction('foo', 'bar');
       expect(jasmineUnderTest.matchersUtil.contains(capturedArgs, 'bar')).toBe(true);
     });
+
+    it("passes for set members", function() {
+      jasmine.getEnv().requireFunctioningSets();
+
+      var setItem = {'foo': 'bar'};
+      var set = new Set();
+      set.add(setItem);
+
+      expect(jasmineUnderTest.matchersUtil.contains(set, setItem)).toBe(true);
+    });
+
+    // documenting current behavior
+    it("fails (!) for objects that equal to a set member", function() {
+      jasmine.getEnv().requireFunctioningSets();
+
+      var set = new Set();
+      set.add({'foo': 'bar'});
+
+      expect(jasmineUnderTest.matchersUtil.contains(set, {'foo': 'bar'})).toBe(false);
+    });
```

Comment:
This is kind of confusing considering arrays are treated differently:
```javascript
expect([{'foo': 'bar'}]).toContain({'foo': 'bar'});    // passes
```
vs
```javascript
expect(new Set([{'foo': 'bar'}])).toContain({'foo': 'bar'});    // fails
```

Reply:
Agreed, and it would be nice to change that but it'll have to go with 4.0.